### PR TITLE
Support list of dict-s as output of a dataset build function

### PR DESCRIPTION
**Pull request by aleksmitov**
To support https://www.notion.so/layerco/SDK-Devex-v1-0ee60206f0df4b70a98e379a60bc45ba#d2049beb08e940b98cf2cda25ae70f3a

After this gets merged, I'll follow up with a PR that uses this logic on the backend executors side.

### CODE REVIEW

File: layer/utils/runtime_utils.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+            raise Exception("pandas.DataFrame object has 0 columns")
+        return result
+    if isinstance(result, list):
+        dict_elements = all([isinstance(list_element, dict) for list_element in result])
+        if not dict_elements:
+            raise Exception("Expected a list of dict elements")
+        return pd.DataFrame.from_records(result)  # type: ignore
+    raise Exception(
+        f"Unsupported return type. Expected one of {allowed_types}, got: {type(result)}"
+    )
```
Comment (cihati) [inline layer/utils/runtime_utils.py:23]: why is this needed?

File: layer/utils/runtime_utils.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
+from typing import Any
```
Comment (cihati) [inline layer/utils/runtime_utils.py:1]: Will this file be used in the executors? If not, perhaps I'd move it under `layer/decorators/` folder.

**Review by cihati — APPROVED** [review]

**Review by cihati — COMMENTED** [review]
Also, please update the docs of dataset_decorator.py

File: layer/utils/runtime_utils.py (hunk 1 above)
Comment (aleksmitov) [reply layer/utils/runtime_utils.py:23]: Linter complains otherwise

File: layer/utils/runtime_utils.py (hunk 2 above)
Comment (aleksmitov) [reply layer/utils/runtime_utils.py:1]: The intention is to be used by an executor, yes.
